scheds/rust/scx_layered/src/main.rs

```diff
@@ -1260,15 +1260,30 @@ impl<'a> Scheduler<'a> {
             let mut core_order: Vec<usize> = core_span.into_iter().collect();
 
             // Shuffle them so that different CPUs follow different orders.
-            // This isn't ideal as random shuffling won't give us complete
-            // fairness. Can be improved by making each CPU radiate in both
-            // directions. For shuffling, use predictable seeds so that
-            // orderings are reproducible.
-            fastrand::seed(cpu_id as u64);
-            fastrand::shuffle(&mut sys_order);
-            fastrand::shuffle(&mut node_order);
-            fastrand::shuffle(&mut llc_order);
-            fastrand::shuffle(&mut core_order);
+            // Each CPU radiates in both directions based on id.
+
+            let radiate = |mut vec: Vec<usize>, x: usize| -> Vec<usize> {
+                if vec.len() <= 1 {
+                    return vec;
+                }
+                vec.sort_by_key(|&num| {
+                    let diff = if num < x { x - num } else { num - x };
+                    (diff, num < x)
+                });
+                let mut result = Vec::new();
+                let left: Vec<usize> = vec[0..vec.len() / 2].iter().cloned().rev().collect();
+                let right: Vec<usize> = vec[vec.len() / 2..vec.len()].iter().cloned().collect();
```

Comment:
If it's already sorted by abs, I'm not sure what the lef/right interposing achieves. The "center" the order is radiating form is already at the beginning of the list, right?

Reply:
I guess it's the difference between:
```
23:01:54 [DEBUG] CPU[79] proximity map[2/40/40/80]: [79, 39, 78, 38, 77, 37, 76, 36, 75, 35, 74, 34, 73, 33, 72, 32, 71, 31, 70, 30, 69, 29, 68, 28, 67, 27, 66, 26, 65, 25, 64, 24, 63, 23, 62, 22, 61, 21, 60, 20, 59, 19, 58, 18, 57, 17, 56, 16, 55, 15, 54, 14, 53, 13, 52, 12, 51, 11, 50, 10, 49, 9, 48, 8, 47, 7, 46, 6, 45, 5, 44, 4, 43, 3, 42, 2, 41, 1, 40, 0]
```
vs
```
23:03:01 [DEBUG] CPU[79] proximity map[2/40/40/80]: [79, 39, 78, 77, 76, 75, 74, 73, 72, 71, 70, 69, 68, 67, 66, 65, 64, 63, 62, 61, 60, 38, 37, 36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 59, 58, 57, 56, 55, 54, 53, 52, 51, 50, 49, 48, 47, 46, 45, 44, 43, 42, 41, 40, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
```